airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -344,7 +365,16 @@ private static Runnable getReplicationRunnable(final AirbyteSource source,
         }
         LOGGER.info("Total records read: {} ({})", recordsRead, FileUtils.byteCountToDisplaySize(messageTracker.getTotalBytesEmitted()));
         if (!validationErrors.isEmpty()) {
+          final TrackingClient trackingClient = TrackingClientSingleton.get();
           validationErrors.forEach((stream, errorPair) -> {
+            if (workspaceId != null) {
```

Comment:
as far as I can tell, `workspaceId` never changes, so this should be able to be combined with the `validationErrors.isEmpty()` check on line 367 to simplify things.